docs/documentation/automation/question_exec.md

```diff
@@ -6,4 +6,105 @@ sidebar_label: Question Exec
 import useBaseUrl from '@docusaurus/useBaseUrl'; 
 import Highlight from '@theme/Highlight';
 
-# Subtitle {#subtitle}
+<span class="hero__subtitle">
+
+_Customize Survey Questions with Javascript Code_
+
+</span>
+
+## Overview
+Surveys are among Cotalker's main features. Their flexibility and customization ability allow them to adapt to each business's needs.
+
+This spec expands the notion of survey functionality by introducing _Question EXEC_, custom code execution for your survey questions.
+
+:::info
+- Custom code must be written in **Javascript**.
+- Code should always be previously _sandboxed_.
+:::
+
+## Set Up
+
+To add your custom code, choose any survey component and select the **Automation** tab. The code is added in the field at the bottom of the settings panel.
+_See the image below:_
+
+<img alt="automation tab" class="img_sizing_narrow item shadow--md" src={useBaseUrl('img/automation_question_exec_00.png')} />
+<br/>
+<br/>
+
+You can add custom code in any of the four _stages_ of a _survey component's lifecycle_. Set the **stage** either with the dropdown menu or by clicking the corresponding icon. 
+_See the image below:_
+
+<img alt="automation tab" class="img_sizing_narrow item shadow--md" src={useBaseUrl('img/automation_question_exec_01.png')} />
+<br/>
+
+## Component Lifecycle Stages
+- **Preload**: The code executes when a new survey instance is summoned.
+- **onDisplay**: Code executes when a survey is in editing mode.
+- **Validate**: Code executes before internal validations, which permits adding additional levels of validation (business guidelines, protocols, etc.).
+- **Postsave**: Code executes after custom and internal validations.
```

Comment:
This runs after it's send to the server??

Reply:
The wiki says: 

> # [postsave: Array<COTCommand<null>>](https://github.com/Cotalker/cotalker-development/wiki/QuestionExec-Specs#postsave-arraycotcommandnull)

> Postsave must be run in question order AFTER the save button is pressed and NO validations in the entire survey fail. The server response is NOT required in MOBILE apps, but IS required in WEB (Mobile Apps fallback to offline mode and guarantee eventual messasge saving)

I'll add this somehow.

Reply:
Instead of "postsave", it's literally a "post-send", right?

Reply:
@Mauriceac yes!
I was worried about this not being really implemented and was confused with `presave: Array<COTCommand<null>>`